RadixWallet/Features/FactorSourceAccess/FactorSourceAccess.swift

```diff
@@ -4,33 +4,110 @@ public struct FactorSourceAccess: Sendable, FeatureReducer {
 		public let kind: Kind
 		public let purpose: Purpose
 
+		@PresentationState
+		public var destination: Destination.State? = nil
+
 		public init(kind: Kind, purpose: Purpose) {
 			self.kind = kind
 			self.purpose = purpose
 		}
 	}
 
-	public enum ViewAction: Sendable, Equatable {
+	public enum ViewAction: Sendable, Hashable {
 		case onFirstTask
 		case retryButtonTapped
 		case closeButtonTapped
 	}
 
-	public enum DelegateAction: Sendable, Equatable {
+	public enum InternalAction: Sendable, Hashable {
+		case isConnectedToAnyConnectorExtension(Bool)
+	}
+
+	public enum DelegateAction: Sendable, Hashable {
 		case perform
 		case cancel
 	}
 
+	public struct Destination: DestinationReducer {
+		@CasePathable
+		public enum State: Sendable, Hashable {
+			case noP2PLink(AlertState<NoP2PLinkAlert>)
+		}
+
+		@CasePathable
+		public enum Action: Sendable, Hashable {
+			case noP2PLink(NoP2PLinkAlert)
+		}
+
+		public var body: some ReducerOf<Self> {
+			EmptyReducer()
+		}
+
+		public enum NoP2PLinkAlert: Sendable, Hashable {
+			case okTapped
+		}
+	}
+
+	@Dependency(\.ledgerHardwareWalletClient) var ledgerHardwareWalletClient
+
+	public var body: some ReducerOf<Self> {
+		Reduce(core)
+			.ifLet(destinationPath, action: /Action.destination) {
+				Destination()
+			}
+	}
+
 	public init() {}
 
-	public func reduce(into _: inout State, viewAction: ViewAction) -> Effect<Action> {
+	private let destinationPath: WritableKeyPath<State, PresentationState<Destination.State>> = \.$destination
+
+	public func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
 		switch viewAction {
-		case .onFirstTask, .retryButtonTapped:
+		case .onFirstTask:
+			.send(.delegate(.perform))
+				.merge(with: checkP2PLinksEffect(state: state))
+		case .retryButtonTapped:
 			.send(.delegate(.perform))
 		case .closeButtonTapped:
 			.send(.delegate(.cancel))
 		}
 	}
+
+	public func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
+		switch internalAction {
+		case let .isConnectedToAnyConnectorExtension(isConnected):
+			if !isConnected {
+				state.destination = .noP2PLink(.noP2Plink)
+			}
+			return .none
+		}
+	}
+
+	public func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
+		switch presentedAction {
+		case .noP2PLink(.okTapped):
+			state.destination = nil
+			return .run { send in
+				// Dispatching this in an async process is enough for it to take place after alert has been dismissed.
+				// No need to place an actual delay with continuousClock.
+				await send(.delegate(.cancel))
+			}
+		}
+	}
+
+	private func checkP2PLinksEffect(state: State) -> Effect<Action> {
+		guard case .ledger = state.kind else {
+			return .none
+		}
+		return .run { send in
+			for try await isConnected in await ledgerHardwareWalletClient.isConnectedToAnyConnectorExtension() {
```

Comment:
I don't think this is what we should check. This can return false if there are P2PLinks, but these are not connected(you browser is closed), I believe that the intention was to inform users when there are no P2PLinks at all.

Reply:
right, will update it to check if there are any p2pLinks instead. 

Do we want to leave it this way when [adding a new Ledger device](https://github.com/radixdlt/babylon-wallet-ios/blob/main/RadixWallet/Features/LedgerHardwareDevices/LedgerHardwareDevices%2BReducer.swift#L206-L210)? Because in such case we will show this alert if there is no active connection, and user could be lead to add a new one when all they need to do is open their browser